server/src/main/java/com/pepperkick/ems/server/entity/Employee.java

```diff
@@ -0,0 +1,157 @@
+package com.pepperkick.ems.server.entity;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+import org.hibernate.annotations.*;
+import org.springframework.lang.Nullable;
+
+import javax.persistence.*;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.util.Comparator;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+@Entity
+@Table(name = "EMPLOYEE")
+@ApiModel
+public class Employee implements Comparable<Employee>, Comparator<Employee> {
+    @Id
+    @Column(name = "ID")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @ApiModelProperty(notes = "Employee's ID", position = 1)
+    private Integer id;
+
+    @Column(name = "NAME", length = 40)
+    @NotNull
+    @ApiModelProperty(notes = "Employee's Name", position = 2, required = true)
+    private String name;
+
+    @Transient
+    @ApiModelProperty(notes = "Employee's Job Title", example = "Director", position = 3, required = true)
+    private String jobTitle;
+
+    @OneToOne
+    @JoinColumn(name = "MANAGER")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @JsonIgnoreProperties(value = {"manager", "colleagues", "subordinates"})
+    @ApiModelProperty(notes = "Employee's Manager", position = 4)
+    private Employee manager;
+
+    @Transient
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @JsonIgnoreProperties(value = {"manager", "colleagues", "subordinates"})
+    @SortComparator(Employee.class)
+    @ApiModelProperty(notes = "Employee's Colleagues", position = 5)
+    private SortedSet<Employee> colleagues;
```

Comment:
the reason you have a comparator and sortedset in this class because your entity is doing too many things. Maintaining separate classes for request/response objects, separate queries for fetching subordinates will make your code align with single responsibility principle.